vault/login_mfa.go

```diff
@@ -106,8 +106,14 @@ func (b *SystemBackend) loginMFAPaths() []*framework.Path {
 	}
 }
 
-func genericOptionalUUIDRegex(name string) string {
-	return fmt.Sprintf("(/(?P<%s>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}))?", name)
+// uuidRegex crafts a regex for use in URL paths, somewhat similar to framework.GenericNameRegex, but only accepting
+// UUIDs, and only lowercase UUIDs at that.
+// It is currently exclusively used for the method_id parameter for MFA methods.
+// Think twice before making use of it in any other context, as restricting the valid input in the URL regex results in
+// an "unsupported path" error, given input which does not match the regex, which is a fairly unclear way to report an
+// invalid parameter value, unless the person seeing the error has an excellent understanding of Vault URL routing.
+func uuidRegex(name string) string {
+	return fmt.Sprintf("(?P<%s>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})", name)
```

Comment:
I disagree with the suggested change, as it implies the parameter value should be a UUID itself. That's not the case, it's the name of a parameter which will contain a UUID. In fact, this function is **exclusively** invoked with the string `"method_id"` so it may be clearer to just turn it into a string constant.

I've already added a documentation comment to it which discourages future re-use in other contexts - removing the parameterization would go a step further to ensure that.

Reply:
Fair point, thanks for clarifying!